### ISSUE TITLE
fix test "test render with bufferOutput=true" .. "wrong buffer" error

### ISSUE THREAD

**Issue by isao** [YahooArchive/mojito#477]
fix errors (intermittent, and randomly on either nodejs 0.6 or 0.8) like [this one](http://travis-ci.org/#!/yahoo/mojito/jobs/2297055).

```
FAILURE DETAILS:
================
mojito-mu-tests :: mojito-mu server tests :: test render with bufferOutput=true
wrong buffer
Expected: <div>bar</div><div>bar</div><div>bar</div> (string)
Actual: <div>bar</div><div>bar</div> (string)
```

**Comment by mridgway** [comment]
I would prefer to just merge #367 and be done with Mu altogether.

**Comment by isao** [comment]
any reason not to merge #367?  is the failing test related to view renderer code changes?

**Comment by mridgway** [comment]
The consensus of the team was to hold off on this until we know that Handlebars is working properly. We should get a new consensus on this and depending on the outcome either fix the Mu view engine or merge #367 and remove Mu entirely.

**Comment by isao** [comment]
fixed with c2a0ad9d439c8b536247ab51bf3b6bb481ae3a0a and 1be70b03a1bf80f0b4c325624a8a5e43a2ee46d8
thanks @mridgway

**Comment by mridgway** [comment]
It should be noted that the tests are now correct, but there is still a race condition in the Mu view renderer when using ac.flush(). Many times the done() call will happen before the flush() completes which is the test failure we were seeing.
